src/overlays/actors/ovl_En_Hata/z_en_hata.c

```diff
@@ -28,16 +28,74 @@ const ActorInit En_Hata_InitVars = {
     (ActorFunc)EnHata_Draw,
 };
 
-#endif
+void EnHata_Init(Actor* thisx, GlobalContext* globalCtx) {
+    EnHata* this = THIS;
+    s32 rand;
+    f32 endFrame;
 
-extern UNK_TYPE D_06002FD0;
+    SkelAnime_Init(globalCtx, &this->skelAnime, &object_hata_Skel_002FD0, NULL, this->jointTable, this->morphTable,
+                   OBJECT_HATA_LIMB_MAX);
+    endFrame = Animation_GetLastFrame(&object_hata_Anim_000444);
+    Animation_Change(&this->skelAnime, &object_hata_Anim_000444, 1.0f, 0.0f, endFrame, ANIMMODE_LOOP, 0.0f);
+    rand = Rand_ZeroFloat(endFrame);
+    this->skelAnime.curFrame = rand;
+    DynaPolyActor_LoadMesh(globalCtx, &this->dyna, &object_hata_Colheader_0000C0);
+    Actor_SetScale(&this->dyna.actor, 0.013f);
+    this->dyna.actor.uncullZoneScale = 500.0f;
+    this->dyna.actor.uncullZoneDownward = 500.0f;
+    this->dyna.actor.uncullZoneForward = 2200.0f;
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Hata/EnHata_Init.s")
+void EnHata_Destroy(Actor* thisx, GlobalContext* globalCtx) {
+    EnHata* this = THIS;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Hata/EnHata_Destroy.s")
+    DynaPoly_DeleteBgActor(globalCtx, &globalCtx->colCtx.dyna, this->dyna.bgId);
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Hata/EnHata_Update.s")
+void EnHata_Update(Actor* thisx, GlobalContext* globalCtx2) {
+    GlobalContext* globalCtx = globalCtx2;
+    EnHata* this = THIS;
+    Vec3f sp34;
+    f32 phi_fv0;
+    s32 pad;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Hata/func_8089EC68.s")
+    phi_fv0 = CLAMP(globalCtx->envCtx.windSpeed / 120.0f, 0.0f, 1.0f);
```

Comment:
Ordinarily I'd say define where declared, but it feels like it makes more sense in this block.